special-pages/pages/history/app/HistoryProvider.js

```diff
@@ -112,9 +114,35 @@ export function HistoryServiceProvider({ service, initial, children }) {
         };
         document.addEventListener('auxclick', handleAuxClick);
 
+        function contextMenu(event) {
+            const target = /** @type {HTMLElement|null} */ (event.target);
+            if (!(target instanceof HTMLElement)) return;
+
+            const actions = {
+                '[data-section-title]': (elem) => elem.querySelector('button')?.value,
+            };
+
+            for (const [selector, valueFn] of Object.entries(actions)) {
+                const match = event.target.closest(selector);
+                if (match) {
+                    const value = valueFn(match);
+                    if (value) {
+                        event.preventDefault();
+                        event.stopImmediatePropagation();
+                        // eslint-disable-next-line promise/prefer-await-to-then
+                        service.menuTitle(value).catch(console.error);
+                    }
+                    break;
+                }
+            }
+        }
+
+        document.addEventListener('contextmenu', contextMenu);
+
```

Comment:
These verbose handlers will be refactored later. Right now I'm trying to capture all requirements first - with tests. Afterwards I'll look at a nicer way to handle the duplication in this file.